SocketIOClientSwift/SocketIOClient.swift

```diff
@@ -175,42 +146,43 @@ public final class SocketIOClient: NSObject, SocketEngineClient, SocketLogClient
     /**
     Connect to the server. If we aren't connected after timeoutAfter, call handler
     */
-    public func connect(#timeoutAfter:Int, withTimeoutHandler handler:(() -> Void)?) {
-        if closed {
-            SocketLogger.log("Warning! This socket was previously closed. This might be dangerous!", client: self)
-            _closed = false
-        } else if connected {
-            return
-        }
-        
-        _connecting = true
-        addEngine()
-        engine?.open(opts: connectParams)
-        
-        if timeoutAfter == 0 {
-            return
-        }
-        
-        let time = dispatch_time(DISPATCH_TIME_NOW, Int64(timeoutAfter) * Int64(NSEC_PER_SEC))
-        
-        dispatch_after(time, dispatch_get_main_queue()) {[weak self] in
-            if let this = self where !this.connected {
-                this._closed = true
-                this._connecting = false
-                this.engine?.close(fast: true)
-                
-                handler?()
+    public func connect(timeoutAfter timeoutAfter:Int,
```

Comment:
Maybe you want to add `assert(timeoutAfter >= 0)` to make sure users don't pass negative values